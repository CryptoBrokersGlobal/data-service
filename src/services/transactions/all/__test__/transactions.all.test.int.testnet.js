const createService = require('..');
const { createPgDriver } = require('../../../../db');
const { parseDate } = require('../../../../utils/parseDate');
const Cursor = require('../../../../resolvers/pagination/cursor');

const TX_ID = '8W9BkioPSWmPfDjcTFGaCy8vLEmcwkzJeSWno1s3Wra7'; // tx 4
const TX_ID_2 = 'CPf7EWE4hPrBNKJpzBtBm9os4UsyZ8Eebhzwq4EqLWqG'; // tx 7
const TX_ID_3 = '23sjEq5zNctBTGqrsapLrPxDkHFM8rJCKF1ti55NRpbF'; // tx 12

const loadConfig = require('../../../../loadConfig');
const options = loadConfig();

const drivers = {
  pg: createPgDriver(options),
};

const service = createService({
  drivers,
  emitEvent: () => () => null,
});

describe('All transactions resolver for one', () => {
  it('fetches real transfer tx', async done => {
    service
      .get(TX_ID)
      .run()
      .promise()
      .then(() => done())
      .catch(e => done(JSON.stringify(e)));
  });
  it('fetches real exchange tx', async done => {
    service
      .get(TX_ID_2)
      .run()
      .promise()
      .then(() => done())
      .catch(e => done(JSON.stringify(e)));
  });
  it('fetches real data tx', async done => {
    service
      .get(TX_ID_3)
      .run()
      .promise()
      .then(() => done())
      .catch(e => done(JSON.stringify(e)));
  });
  it('returns null for unreal tx', async () => {
    const tx = await service
      .get('UNREAL')
      .run()
      .promise();

    expect(tx).toBe(null);
  });
});

describe('All transactions resolver for many', () => {
  it('fetches real tx', async () => {
    const txs = await service
      .search({ limit: 20, sort: 'asc' })
      .run()
      .promise();

    expect(txs).toBeDefined();
    expect(txs.data).toHaveLength(20);
    expect(txs).toMatchSnapshot();
  });

  describe('Pagination ', async () => {
    const START = '2018-07-01T00:00:00.000Z';
    const END = '2018-10-01T00:00:00.000Z';
    const LIMIT = 21;
    const createCursor = sort => ({ data }) => Cursor.encode(sort, data);
    it('doesnt get 2 identical entries for limit 1 asc with next page fetching', async () => {
      const baseParams = {
        limit: 1,
        timeStart: parseDate(START),
        sort: 'asc',
      };

      const firstTx = await service
        .search(baseParams)
        .run()
        .promise();

      const secondTx = await service
        .search({
          after: firstTx.lastCursor,
          limit: 1,
        })
        .run()
        .promise();

      expect(firstTx.data).not.toEqual(secondTx.data);
    });
    it(' works asc', async () => {
      const SORT = 'asc';

      const baseParams = {
        limit: LIMIT,
        timeEnd: parseDate(END),
        timeStart: parseDate(START),
        sort: SORT,
      };

      const fetchAndGetNextCursor = cursor =>
        service
          .search({
            ...baseParams,
            limit: 5,
            after: cursor,
          })
          .run()
          .promise()
          .then(x => [x.lastCursor, x.data.map(createCursor(SORT))]);

      const firstCursor = await service
        .search({ ...baseParams, limit: 1 })
        .run()
        .promise()
        .then(x => x.data.map(createCursor(SORT))[0]);
      var i = 0;
      var cursors = [firstCursor];
      var curCursor = firstCursor;

      while (i++ < (LIMIT - 1) / 5) {
        var [nextCursor, cs] = await fetchAndGetNextCursor(curCursor);
        curCursor = nextCursor;
        cursors = [...cursors, ...cs];
      }

      const expectedCursors = await service
        .search({
          ...baseParams,
          limit: LIMIT,
        })
        .run()
        .promise()
        .then(x => x.data.map(createCursor(SORT)));

      expect(cursors).toEqual(expectedCursors);
    });
    it(' works desc', async () => {
      const SORT = 'desc';

      const baseParams = {
        limit: LIMIT,
        timeEnd: parseDate(END),
        timeStart: parseDate(START),
        sort: SORT,
      };

      const fetchAndGetNextCursor = cursor =>
        service
          .search({
            ...baseParams,
            limit: 5,
            after: cursor,
          })
          .run()
          .promise()
          .then(x => [x.lastCursor, x.data.map(createCursor(SORT))]);

      const firstCursor = await service
        .search({ ...baseParams, limit: 1 })
        .run()
        .promise()
        .then(x => x.data.map(createCursor(SORT))[0]);
      var i = 0;
      var cursors = [firstCursor];
      var curCursor = firstCursor;

      while (i++ < (LIMIT - 1) / 5) {
        var [nextCursor, curCursors] = await fetchAndGetNextCursor(curCursor);
        curCursor = nextCursor;
        cursors = [...cursors, ...curCursors];
      }

      const expectedCursors = await service
        .search({
          ...baseParams,
          limit: LIMIT,
        })
        .run()
        .promise()
        .then(x => x.data.map(createCursor(SORT)));

      expect(cursors).toEqual(expectedCursors);
    });
  });

  it('fails if timeEnd < 0', done =>
    service
      .search({
        timeEnd: parseDate('-1525132900000'),
      })
      .run()
      .promise()
      .then(() => done('Wrong branch, error'))
      .catch(e => {
        expect(e.type).toBe('ValidationError');
        done();
      }));
  it('fails if timeStart < 0', done =>
    service
      .search({
        timeEnd: parseDate('1525132900000'),
        timeStart: parseDate('-1525132800000'),
      })
      .run()
      .promise()
      .then(() => done('Wrong branch, error'))
      .catch(e => {
        expect(e.type).toBe('ValidationError');
        done();
      }));
  it('fails if timeEnd < timeStart', done =>
    service
      .search({
        timeEnd: parseDate('1525132700000'),
        timeStart: parseDate('1525132800000'),
      })
      .run()
      .promise()
      .then(() => done('Wrong branch, error'))
      .catch(e => {
        expect(e.type).toBe('ValidationError');
        done();
      }));
  it('fails if timeStart->invalid Date', done =>
    service
      .search({
        timeStart: parseDate(''),
      })
      .run()
      .promise()
      .then(() => done('Wrong branch, error'))
      .catch(e => {
        expect(e.type).toBe('ValidationError');
        done();
      }));

  it('doesnt try to create a cursor for empty response', done =>
    service
      .search({
        limit: 1,
        timeEnd: parseDate('1'),
      })
      .run()
      .promise()
      .then(d => {
        expect(d).not.toHaveProperty('lastCursor');
        done();
      })
      .catch(e => done(JSON.stringify(e, null, 2))));
});
