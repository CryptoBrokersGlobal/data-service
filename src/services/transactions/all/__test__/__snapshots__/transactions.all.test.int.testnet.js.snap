// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All transactions resolver for many fetches real tx 1`] = `
Object {
  "__type": "list",
  "data": Array [
    Object {
      "__type": "transaction",
      "data": Object {
        "amount": "2000000",
        "fee": "0",
        "height": 1,
        "id": "3obfFPvsWXv2RyMYxjTT7owYGcpSGuSAm8fQVXeX5wErWYsgNSPPnQoFVV6nzuwm3RwGCbm8dfgvqwK9S8fVMpye",
        "recipient": "3N5GRqzDBhjVXnCn44baHcz2GoZy5qLxtTh",
        "signature": "3obfFPvsWXv2RyMYxjTT7owYGcpSGuSAm8fQVXeX5wErWYsgNSPPnQoFVV6nzuwm3RwGCbm8dfgvqwK9S8fVMpye",
        "timestamp": 2016-11-01T08:33:20.000Z,
        "type": 1,
      },
    },
    Object {
      "__type": "transaction",
      "data": Object {
        "amount": "2000000",
        "fee": "0",
        "height": 1,
        "id": "3TdE9G7V7fwED35981aGsWFM6aesxSS4W1XPfEx6p5xacwHLu7Kvf67Wzg73kgyU9gSFp1KsmPWqkFhaaR2S1fhp",
        "recipient": "3NCBMxgdghg4tUhEEffSXy11L6hUi6fcBpd",
        "signature": "3TdE9G7V7fwED35981aGsWFM6aesxSS4W1XPfEx6p5xacwHLu7Kvf67Wzg73kgyU9gSFp1KsmPWqkFhaaR2S1fhp",
        "timestamp": 2016-11-01T08:33:20.000Z,
        "type": 1,
      },
    },
    Object {
      "__type": "transaction",
      "data": Object {
        "amount": "2000000",
        "fee": "0",
        "height": 1,
        "id": "3zpi4i5SeCoaiCBn1iuTUvCc5aahvtabqXBTrCXy1Y3ujUbJo56VVv6n4HQtcwiFapvg3BKV6stb5QkxsBrudTKZ",
        "recipient": "3NBVqYXrapgJP9atQccdBPAgJPwHDKkh6A8",
        "signature": "3zpi4i5SeCoaiCBn1iuTUvCc5aahvtabqXBTrCXy1Y3ujUbJo56VVv6n4HQtcwiFapvg3BKV6stb5QkxsBrudTKZ",
        "timestamp": 2016-11-01T08:33:20.000Z,
        "type": 1,
      },
    },
    Object {
      "__type": "transaction",
      "data": Object {
        "amount": "90000000",
        "fee": "0",
        "height": 1,
        "id": "4hTrr7fqkujsGSH8AFN1qw7fJdfmKgwzoq3ByCCJwduHkgZPQZe1KgzG6oPBZXMuNr5ZQ6ErDSTiz2KGtxtkHpA5",
        "recipient": "3N18z4B8kyyQ96PhN5eyhCAbg4j49CgwZJx",
        "signature": "4hTrr7fqkujsGSH8AFN1qw7fJdfmKgwzoq3ByCCJwduHkgZPQZe1KgzG6oPBZXMuNr5ZQ6ErDSTiz2KGtxtkHpA5",
        "timestamp": 2016-11-01T08:33:20.000Z,
        "type": 1,
      },
    },
    Object {
      "__type": "transaction",
      "data": Object {
        "amount": "4000000",
        "fee": "0",
        "height": 1,
        "id": "5G66c9GPn2egiM4bQBBF3gCkHS8sQZupRvWCpWKWGQTRRbtqdtZJ5Mt29exbHTDZW2RWygVKZ3oBNg4RwezN7wmA",
        "recipient": "3My3KZgFQ3CrVHgz6vGRt8687sH4oAA1qp8",
        "signature": "5G66c9GPn2egiM4bQBBF3gCkHS8sQZupRvWCpWKWGQTRRbtqdtZJ5Mt29exbHTDZW2RWygVKZ3oBNg4RwezN7wmA",
        "timestamp": 2016-11-01T08:33:20.000Z,
        "type": 1,
      },
    },
    Object {
      "__type": "transaction",
      "data": Object {
        "assetId": "8jfD2JBLe23XtCCSQoTx5eAW5QCU6Mbxi3r78aNQLcNf",
        "decimals": 2,
        "description": "2VKqj",
        "fee": "1",
        "height": 193,
        "id": "8jfD2JBLe23XtCCSQoTx5eAW5QCU6Mbxi3r78aNQLcNf",
        "name": "7a29yz",
        "quantity": "9846148.03",
        "reissuable": false,
        "sender": "3Mv61qe6egMSjRDZiiuvJDnf3Q1qW9tTZDB",
        "senderPublicKey": "FkoFqtAeibv2E6Y86ZDRfAkZz61LwUMjLAP2gmS1j7xe",
        "signature": "2kSNWQM7zx4xXkSZrksZPSNYY32WXAdvQcBTvySMrtZ5rzWYpyXYcm8tbdUSjbyBT2LtMfiyQnYXguxrMdLpWgCq",
        "timestamp": 2017-01-27T11:56:41.965Z,
        "type": 3,
        "version": 1,
      },
    },
    Object {
      "__type": "transaction",
      "data": Object {
        "amount": "10",
        "assetId": "WAVES",
        "attachment": "",
        "fee": "0.001",
        "feeAsset": "WAVES",
        "height": 56,
        "id": "8W9BkioPSWmPfDjcTFGaCy8vLEmcwkzJeSWno1s3Wra7",
        "recipient": "3N78bNBYhT6pt6nugc6ay1uW1nLEfnRWkJd",
        "sender": "3NBVqYXrapgJP9atQccdBPAgJPwHDKkh6A8",
        "senderPublicKey": "CRxqEuxhdZBEHX42MU4FfyJxuHmbDBTaHMhM3Uki7pLw",
        "signature": "4W9nWkfRm62rTQiuZX6bowWmDnz5n3cKhCZmLcYgivK53mBt3TzH6Z52fV6fXPSZn5bZc97rNo76usnNEoQcTHaq",
        "timestamp": 2017-01-27T12:00:37.072Z,
        "type": 4,
        "version": 1,
      },
    },
    Object {
      "__type": "transaction",
      "data": Object {
        "amount": "10",
        "assetId": "WAVES",
        "attachment": "",
        "fee": "0.001",
        "feeAsset": "WAVES",
        "height": 187,
        "id": "Cxk9JyLq9HCuik6cJ4U13ea2pNEopcsfAjyHNXgZeew7",
        "recipient": "3Mv61qe6egMSjRDZiiuvJDnf3Q1qW9tTZDB",
        "sender": "3NBVqYXrapgJP9atQccdBPAgJPwHDKkh6A8",
        "senderPublicKey": "CRxqEuxhdZBEHX42MU4FfyJxuHmbDBTaHMhM3Uki7pLw",
        "signature": "hTK3XVj8rgmGhDYh8vgMU8E3uhR6KypJ6jaSpJjwJt4AYNGnD25yExAVGHL5JgdwPD9i3Fcw6u6TThN1u9XNSuD",
        "timestamp": 2017-01-27T12:03:03.195Z,
        "type": 4,
        "version": 1,
      },
    },
    Object {
      "__type": "transaction",
      "data": Object {
        "amount": "1000",
        "assetId": "WAVES",
        "attachment": "",
        "fee": "1",
        "feeAsset": "WAVES",
        "height": 292,
        "id": "6biG9Fogmi3Md1nEEKMQ8KggRiXzwEH7TyGHXCrxiPnw",
        "recipient": "3Mv61qe6egMSjRDZiiuvJDnf3Q1qW9tTZDB",
        "sender": "3NBVqYXrapgJP9atQccdBPAgJPwHDKkh6A8",
        "senderPublicKey": "CRxqEuxhdZBEHX42MU4FfyJxuHmbDBTaHMhM3Uki7pLw",
        "signature": "2k9U1eHts4PgrGU4tPz6UuU2dgBzTLu2GcBFD1ZiKzeWYZut9mjQ22zy3zhBTUe16nfUa4owYm9FZ1vRA1Lde1Ej",
        "timestamp": 2017-01-27T12:10:09.608Z,
        "type": 4,
        "version": 1,
      },
    },
    Object {
      "__type": "transaction",
      "data": Object {
        "assetId": "Fpdfro5ZQg6PPDhjH4hLnUEVyWGsBkae5acu5oA5iEtR",
        "decimals": 2,
        "description": "2VKqj",
        "fee": "1",
        "height": 299,
        "id": "Fpdfro5ZQg6PPDhjH4hLnUEVyWGsBkae5acu5oA5iEtR",
        "name": "7a29yz",
        "quantity": "1152047.2",
        "reissuable": true,
        "sender": "3Mv61qe6egMSjRDZiiuvJDnf3Q1qW9tTZDB",
        "senderPublicKey": "FkoFqtAeibv2E6Y86ZDRfAkZz61LwUMjLAP2gmS1j7xe",
        "signature": "667dYQWVf17Ajb8hC4aCLy9pC83dtVdbeHvGKs3Bh4TXnpY9zBf9Q4cLQMyxym3v2NuYTgaTeyqMx1Eo9yDH8HHB",
        "timestamp": 2017-01-27T12:11:16.889Z,
        "type": 3,
        "version": 1,
      },
    },
    Object {
      "__type": "transaction",
      "data": Object {
        "amount": "2.5e-7",
        "assetId": "WAVES",
        "attachment": "2Bpuwbv51DzmKftL8NwZv6n5zKCBTs5YSVJv9zFGhUYBswvycfCkxdUSwSBcNhC266AWjFC1SsnqDEGGPTfoMgnGvjZj94A6mEfTiEQGdafao8b5asJpZ1PuhcEctWFfRufh5aKg8Nb5Ps2nv2vJFBNrjdQDxzMcoUPYJGZ71D5QQjVXVXc51EFU1p2VpJ4x",
        "fee": "0.00165404",
        "feeAsset": "WAVES",
        "height": 299,
        "id": "EmNgNB6gj9FLbKddAWFZ6NBBh9rWbgD5YfbVSiQapDuC",
        "recipient": "3N5jhcA7R98AUN12ee9pB7unvnAKfzb3nen",
        "sender": "3Mv61qe6egMSjRDZiiuvJDnf3Q1qW9tTZDB",
        "senderPublicKey": "FkoFqtAeibv2E6Y86ZDRfAkZz61LwUMjLAP2gmS1j7xe",
        "signature": "2x8oGHYBH1W6kpoSnzm15ZrcUYopbDfmBu2FoBKrr72ELfK6rtahXAeTbSFNZSR1wHvnqsPkqMHE8ycN6YjfoseF",
        "timestamp": 2017-01-27T12:11:16.890Z,
        "type": 4,
        "version": 1,
      },
    },
    Object {
      "__type": "transaction",
      "data": Object {
        "amount": "9.9e-7",
        "assetId": "WAVES",
        "attachment": "ciAmdk6ruQ2dh49k2jnE2EXGAeuCj9kR5hhhE7UbJnnAshfyndxXa7JAn6jASx3Uq7B7w8nDcigFFiF9FsmbD9AnXLMVitAz4v6TCuTMFCmSTK36e15UKPmKiWQ2M244mz8Mn71BMyCjzRbSVKGYGQe5iJGyeKH6pK7nw12TBQjFsZnhU8k17YzdfWYkzfS",
        "fee": "0.00167663",
        "feeAsset": "WAVES",
        "height": 299,
        "id": "BNoTCES3rVf3PubRHSdyvQfHetNjw5pZWiE8aAzmEsPS",
        "recipient": "3N5jhcA7R98AUN12ee9pB7unvnAKfzb3nen",
        "sender": "3Mv61qe6egMSjRDZiiuvJDnf3Q1qW9tTZDB",
        "senderPublicKey": "FkoFqtAeibv2E6Y86ZDRfAkZz61LwUMjLAP2gmS1j7xe",
        "signature": "2CwNXRWnh6EFa4UkxUq9if5xYenPMvYwwFidE1yTFL8zegxJDvfBoYFxk1yQkDiih18mjr3xaMTTiP4C6iN5zpym",
        "timestamp": 2017-01-27T12:11:16.894Z,
        "type": 4,
        "version": 1,
      },
    },
    Object {
      "__type": "transaction",
      "data": Object {
        "amount": "3.2e-7",
        "assetId": "WAVES",
        "attachment": "38gAknGL2Hzz9xEaz9qzvgNnQTgftYcYF4pf1FgMAmwt5Nm1w4ZRej7bknnzJzcVQdxiFxwCfAM32utQB2xLQQqFTiiSLDqdEXG1mUBsfuAbEb3Kki7U2KU6VJ2VtUuDcEY1WVvJu8UoatjGHX5HFBzPF6ujXTYJtC8sMo559j1p6Hg6Z3vfshKjsaXj7YHE",
        "fee": "0.00180231",
        "feeAsset": "WAVES",
        "height": 299,
        "id": "8kVJEbUiXixJrRkypRJ8bFyaw9ZAbes7ajtRJZvGvxiJ",
        "recipient": "3N5jhcA7R98AUN12ee9pB7unvnAKfzb3nen",
        "sender": "3Mv61qe6egMSjRDZiiuvJDnf3Q1qW9tTZDB",
        "senderPublicKey": "FkoFqtAeibv2E6Y86ZDRfAkZz61LwUMjLAP2gmS1j7xe",
        "signature": "2kVE4B2A2BzdWPeGnZBwbGGAL7gvsNbdzsUTrxg2h4Jh8LRRv12YnsyAv5TTeZFMKgygQZ8RWhLfgCPXcDfHLppH",
        "timestamp": 2017-01-27T12:11:16.896Z,
        "type": 4,
        "version": 1,
      },
    },
    Object {
      "__type": "transaction",
      "data": Object {
        "amount": "6.4e-7",
        "assetId": "WAVES",
        "attachment": "j4S5rmoCjew1FAompPCDyjNdj3iNh3DLmmhhnpgYfcowNYn55qXstBPEZwr97AkTdJmf8kBf6hTdzmdD2EUebzRjvS2QpSS73ijUcGspizuhaY2ddfLTfo5W3ed2UKKKgLr7dbR7eoVquqKt3UdwcPj73z2hTqswLpir11Vpn31pWjC6hVeUy8H6Dz3oKcJ",
        "fee": "0.00113659",
        "feeAsset": "WAVES",
        "height": 299,
        "id": "Cm7MFpahyGd7g8qrYtmbikKZq8K4EECgPNp7PAea9cx1",
        "recipient": "3Mv61qe6egMSjRDZiiuvJDnf3Q1qW9tTZDB",
        "sender": "3Mv61qe6egMSjRDZiiuvJDnf3Q1qW9tTZDB",
        "senderPublicKey": "FkoFqtAeibv2E6Y86ZDRfAkZz61LwUMjLAP2gmS1j7xe",
        "signature": "5EJuw3djnfuGJUARfj1qRx7N7f5EXEK3K7SZuEuGaUiGZgBKte93jKvghTkmrgK92AtmP2NB5c8dnzoPw6qSc7La",
        "timestamp": 2017-01-27T12:11:16.897Z,
        "type": 4,
        "version": 1,
      },
    },
    Object {
      "__type": "transaction",
      "data": Object {
        "amount": "6e-7",
        "assetId": "WAVES",
        "attachment": "2dPJSpyvGAJgsiAuMR7ZEFsPrDd6VwE8TJFzVqZLMgzYG367nmgqv5S25FqEVT9vGLJY6PsXxSCuFeS2EXc4cWgXKXKMWqSa2bcTAUToSJbM2KvybAs7GDJAoatqrBMsCKGJCXmpYnNiLR22krMcxbd8EazmHFRzeKUPnQV67YzEA3FBEPLvX8tP5eCyXavt",
        "fee": "0.00146609",
        "feeAsset": "WAVES",
        "height": 299,
        "id": "3gykusZFbckBvtkUC9xiTidP9G84DVT4Rxo7bP5JtkPi",
        "recipient": "3Mv61qe6egMSjRDZiiuvJDnf3Q1qW9tTZDB",
        "sender": "3Mv61qe6egMSjRDZiiuvJDnf3Q1qW9tTZDB",
        "senderPublicKey": "FkoFqtAeibv2E6Y86ZDRfAkZz61LwUMjLAP2gmS1j7xe",
        "signature": "q7Kve7JrUwWxMyaHUsNiWossYJMAv8S2p28j28rPANmcabbD9hcAatAwJEv9XtYzvvGJcCNHVJivmSHf4mBfWH3",
        "timestamp": 2017-01-27T12:11:16.902Z,
        "type": 4,
        "version": 1,
      },
    },
    Object {
      "__type": "transaction",
      "data": Object {
        "amount": "6.2e-7",
        "assetId": "WAVES",
        "attachment": "QkqprAhbYh3N3RuqCKGTTnTvCVQzAgavoXcGaw2g2qqrH5MCwwVqL7S9J7BRsGFwZtWhH4Fwa8mEgxTi9gdb92G3dfP1cLfPzuLgYcC5UvgWTqbHJUWBoQRmTXcJ84FHBRdgcTiWv4CYcEcLbezsrdRqf6ZcLYq7oaqEvHRTE6rLPb3EjaMNTgdyjU3Mw6W",
        "fee": "0.00120815",
        "feeAsset": "WAVES",
        "height": 299,
        "id": "6FW1YK7H39UeBRGUTx5fAsM8G2aqoHWMsGXPrHt6knF1",
        "recipient": "3Mv61qe6egMSjRDZiiuvJDnf3Q1qW9tTZDB",
        "sender": "3Mv61qe6egMSjRDZiiuvJDnf3Q1qW9tTZDB",
        "senderPublicKey": "FkoFqtAeibv2E6Y86ZDRfAkZz61LwUMjLAP2gmS1j7xe",
        "signature": "4En5g2wSnaMXDec6Zyn2ayzuJsFczRuPVEmnLAkRhJ5JvdYj7gTtYQAfBU8EGczg2fgWycs8PjpNympRzrSFVc9f",
        "timestamp": 2017-01-27T12:11:16.905Z,
        "type": 4,
        "version": 1,
      },
    },
    Object {
      "__type": "transaction",
      "data": Object {
        "amount": "9.8e-7",
        "assetId": "WAVES",
        "attachment": "2bYgKqkKDC8cswN6LBJh6RaWtF8dpptXe3kntSgndXTnpYDS4kRB6C2hXrDdaJkHscxu9oB7cEZmMCmK3F3T9VWw1gHCayZdhatY37DsAznRKomykvyYD3xqTdWEiciyEuo1PNf9WKZcGbsSmV6tWFQAVKuTCPEWJbbGszcpq9e8adHwhYjs7QCNEeFAXQFx",
        "fee": "0.0013209",
        "feeAsset": "WAVES",
        "height": 299,
        "id": "8EMUEsSzDWoWdNZQueCwNwzGyCC54NGawDchVwMsysfZ",
        "recipient": "3Mv61qe6egMSjRDZiiuvJDnf3Q1qW9tTZDB",
        "sender": "3Mv61qe6egMSjRDZiiuvJDnf3Q1qW9tTZDB",
        "senderPublicKey": "FkoFqtAeibv2E6Y86ZDRfAkZz61LwUMjLAP2gmS1j7xe",
        "signature": "42xk7faBHUuwjiH1ynoV3dvKDwJndTesK31UvzMmzK6LTq6Em9fAvKxxqkAEQVB8eF1nZ15XYNrZP4CLwpRA8Zoz",
        "timestamp": 2017-01-27T12:11:16.908Z,
        "type": 4,
        "version": 1,
      },
    },
    Object {
      "__type": "transaction",
      "data": Object {
        "amount": "4.3e-7",
        "assetId": "WAVES",
        "attachment": "2E7uakKLJ6pCTDoCsTP2bSJXA47DPkSHoswREpU9eCei5RC3qXhdewygvkffsxc66BTpBdA2gVWgVnq7FV7T9ChPY5erSN3zbv48KYQHreWY1RdGYsVGAd9kfjSX6L33oyPJDP2WxqmW91yMCVM5RmTnJTW649uDk6QJsY6B7Fote5PVAdWpse8n1ZmXZ39c",
        "fee": "0.00180942",
        "feeAsset": "WAVES",
        "height": 299,
        "id": "8XxkvM2USjSUEb54hWN5bjds8dy74M23mmj265oRCXTT",
        "recipient": "3N5jhcA7R98AUN12ee9pB7unvnAKfzb3nen",
        "sender": "3Mv61qe6egMSjRDZiiuvJDnf3Q1qW9tTZDB",
        "senderPublicKey": "FkoFqtAeibv2E6Y86ZDRfAkZz61LwUMjLAP2gmS1j7xe",
        "signature": "5fF7TwxxAFYNGNsXjUzVQ1C8MHteC1k5MPSmv2hn5Q9tXkq1eF1srRkyuzk8wSd6QqNENNicQpVz8SjjS4UoudQc",
        "timestamp": 2017-01-27T12:11:16.910Z,
        "type": 4,
        "version": 1,
      },
    },
    Object {
      "__type": "transaction",
      "data": Object {
        "amount": "2e-8",
        "assetId": "WAVES",
        "attachment": "7f2D5F9iSRyZp2RfkqATin7aojfP5P3xBiPuSc4hM3bWgcLJdP1tP1hxKhfyKeaNU2jfi3Vo8YkFubL5vkT7kqMyEwWY8ztup4in4eEGK7jeuamFTdFyNhNQ8vSJJHwWcrsBpdhhBQ7cM1Wu8i18wpsnDSUqsUHxkciF1MYZZhCPRLXwWr6AHBSYohMD8Qx",
        "fee": "0.00107679",
        "feeAsset": "WAVES",
        "height": 299,
        "id": "5V6sYzgtLgBh3GjU3ZG3ucmFvAPmRYACGwerKK2bjkvv",
        "recipient": "3N5jhcA7R98AUN12ee9pB7unvnAKfzb3nen",
        "sender": "3Mv61qe6egMSjRDZiiuvJDnf3Q1qW9tTZDB",
        "senderPublicKey": "FkoFqtAeibv2E6Y86ZDRfAkZz61LwUMjLAP2gmS1j7xe",
        "signature": "pDrALhXPBadMRX26HVsgFiDzPXQGpeqqkNCUXLvPrFCZtpxinXrvGqsUA3Mg6ZJvDiqiC1aT5kWVbuqrtHMDSH8",
        "timestamp": 2017-01-27T12:11:16.913Z,
        "type": 4,
        "version": 1,
      },
    },
    Object {
      "__type": "transaction",
      "data": Object {
        "amount": "8.3e-7",
        "assetId": "WAVES",
        "attachment": "6PeYhdAe3BppZgydvRBks63rTSpeXbSL6ygHmyXNPqhKmBfmPjoWL7h2XRXZEjNS5MtvmYW2MrCPfr8cfimktJCW7RZHLnfpWJwmMVMvPbKLranBRNyQscpJrf1Yh3SqHjqLXXXXLS3n161ae1EJ769NwCyefK3xNJYU1nELAU2VcCaaCBqvq5YWnFy4gX5",
        "fee": "0.00182478",
        "feeAsset": "WAVES",
        "height": 299,
        "id": "7rNtFwna6zGJmoh1Yp3E6iprBG5M2v3nt8jkze3ocnmS",
        "recipient": "3N5jhcA7R98AUN12ee9pB7unvnAKfzb3nen",
        "sender": "3Mv61qe6egMSjRDZiiuvJDnf3Q1qW9tTZDB",
        "senderPublicKey": "FkoFqtAeibv2E6Y86ZDRfAkZz61LwUMjLAP2gmS1j7xe",
        "signature": "3DJqu6zW8UiDKsHBmxCGRLpDSmdh8Bf4TwpCmbuwKByYXAkpWkUt33qukCszibzt8mc4F7qBx1HKUePf8Fhtm1GG",
        "timestamp": 2017-01-27T12:11:16.922Z,
        "type": 4,
        "version": 1,
      },
    },
  ],
  "lastCursor": "MjAxNy0wMS0yN1QxMjoxMToxNi45MjJaOjo3ck50RnduYTZ6R0ptb2gxWXAzRTZpcHJCRzVNMnYzbnQ4amt6ZTNvY25tUzo6YXNj",
}
`;
